File: backend/src/main/java/ru/itmo/eduassistant/backend/config/ApiConfig.java
package ru.itmo.eduassistant.backend.config;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestTemplate;
import ru.itmo.eduassistant.commons.client.telegram.TelegramMessageClient;

@Configuration
@RequiredArgsConstructor
public class ApiConfig {
    private final RestTemplate restTemplate;
    @Value("${backend.bot-url}")
    private String telegramUrl;

    @Bean
    TelegramMessageClient telegramClient() {
        return new TelegramMessageClient(restTemplate, telegramUrl);
    }
}

File: bot/src/main/kotlin/ru/itmo/eduassistant/bot/config/ApiConfig.kt
package ru.itmo.eduassistant.bot.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestTemplate
import ru.itmo.eduassistant.commons.client.backend.EduAssistantClient

@Configuration
class ApiConfig(private val restTemplate: RestTemplate, @Value("\${bot.backend-url}") private val backendUrl: String) {

    @Bean
    fun eduAssistantClient(): EduAssistantClient {
        return EduAssistantClient(restTemplate, backendUrl);
    }
}